monailabel/tasks/train/basic_train.py

```diff
@@ -48,29 +59,34 @@
 
 class Context:
     def __init__(self):
-        self.run_id = None
-        self.output_dir = None
-        self.events_dir = None
-        self.train_ds = None
-        self.train_batch_size = None
-        self.val_ds = None
-        self.val_batch_size = None
-        self.publisher = None
-        self.device = None
-        self.network = None
-        self.dataset_type = "CacheDataset"
-        self.pretrained = False
-        self.max_epochs = 1
-        self.multi_gpu = False
-        self.local_rank = 0
-        self.world_size = 0
+        self.start_ts = 0  # timestamp
+        self.run_id = None  # unique run_id
+        self.output_dir = None  # output dir for storing model
+        self.events_dir = None  # events dir for storing tensorboard events
+        self.datalist = None  # input datalist
+        self.train_datalist = None  # train datalist
+        self.train_batch_size = None  # train batch size
+        self.val_datalist = None  # validation datalist
+        self.val_batch_size = None  # validation batch size
+        self.device = None  # device on which training will run
+        self.network = None  # network
+        self.dataset_type = "CacheDataset"  # dataset type
+        self.dataloader_type = "ThreadDataLoader"  # dataloader type
```

Comment:
This will make the ThreadDataLoader by default which as you said can be quite taxing

Reply:
Yes.. changed the mind.. Best as default.. and options for people who own lower resources.. another way to influence them to upgrade to better hardware.. but they will have option to run without code changes on lower resources